fenics_ice/model.py

```diff
@@ -148,7 +148,17 @@ def alpha_from_data(self):
     def bglen_from_data(self):
         """Get bglen field from initial input data"""
         self.bglen = self.input_data.interpolate("Bglen", self.Q)
-
+        """Get bglen mask field from input data"""
+        self.bglen_mask = self.input_data.interpolate("Bglenmask", self.M)
+
+        """bglen_mask is currently just bglen where data available and nan elsehere"""
+        """Following is hacky way to convert this to 0 where nan, 1 elsewhere"""
+        temp_vec = self.bglen_mask.vector()[:]
+        temp_vec[np.isnan(temp_vec)] = 0.0
+        temp_vec[temp_vec>0.0]=1.0
+        self.bglen_mask.vector()[:] = temp_vec
+        """Following appears in inout.interpolate -- is something similar needed?"""
+        self.bglen_mask.vector().apply("insert")
```

Comment:
```
self.bglen_mask.set_local(temp_vec)
self.bglen_mask.vector().apply("insert")
```
or can use the tlm_adjoint interface, `function_set_local(self.bglen_mask, temp_vec)`.

Reply:
@jrmaddison  see new changes to this function. i still use the "insert" argument for `apply`, is this appropriate?

Reply:
I'm not sure `apply` is needed here, but it seems safer to leave it.